### PR TITLE
Video autoplay #3

### PR DESCRIPTION
**Pull request by glebmachine**
VideoBaseTexture now have autoplay parameter, default `true`
I hope this PR is correct)

### CODE REVIEW

**Comment by englercj** [comment]
Looks good, just a couple notes:

1. It would be nice to set this via the constructor. Mind adding a param (and associated jsdoc)?
2. Please add the newline at the end of the file back.

Thanks!

**Comment by glebmachine** [comment]
@englercj here you go! ^_^

File: src/core/textures/VideoBaseTexture.js [hunk 1]
```diff
@@ -54,6 +55,15 @@ function VideoBaseTexture(source, scaleMode)
      */
     this.autoUpdate = false;
 
+    /**
+     * Should video automatically played on "onCanPlay" event
+     *
+     * @member {boolean}
+     * @default true
+     */
+    console.log('---->', source, autoPlay);
```
Comment (englercj) [inline src/core/textures/VideoBaseTexture.js]: Oops!

**Comment by glebmachine** [comment]
oh shi)

**Comment by GoodBoyDigital** [comment]
Hey @glebmachine think we should set it to autoplay by default? Mainly as its how v3 worked. cc @englercj

**Comment by glebmachine** [comment]
Hm, no it shouldn't.

**Comment by glebmachine** [comment]
Will fix that soon!

**Comment by englercj** [comment]
@GoodBoyDigital Looks like it does auto play by default: https://github.com/pixijs/pixi.js/pull/2241/files#diff-c4e302809da411c848ddeebf098e3391R64

**Comment by GoodBoyDigital** [comment]
oh! my bad! I have been staring at a lot of code today :)

**Comment by glebmachine** [comment]
Done

**Comment by englercj** [comment]
I don't know why but you changed `autoplay` to default to `false`, which we don't want, and merged in master, which we don't want.

You're PR was about to be merged, not sure why you made these changes?

**Comment by glebmachine** [comment]
> Hm, not sure, just about to keep branch up to date..

I thought you wanna made it default false, don't you!

**Comment by glebmachine** [comment]
Will recreate PR later

**Comment by GoodBoyDigital** [comment]
Hey @glebmachine sorry about the confusion here! Autoplay being true by default is the way to go (and the way you had it originally) The main reason for this is to keep it in line with how v3 currently works. 

Apologies again for the mix up. I will now send my self back to code reading school...

**Comment by ivanpopelyshev** [comment]
I think its better to make autoplay to false by default.

**Comment by photonstorm** [comment]
Based on Apple announcements today, Safari on iOS will now autoplay videos, as long as they don't have an audio track. It also now supports inline videos, without needing them to go fullscreen. Both of which are great really, and finally allows for videos as textures on iOS, not just on desktop.

https://developer.apple.com/library/prerelease/content/releasenotes/General/WhatsNewInSafari/Articles/Safari_10_0.html

In my mind, I'd expect a video to play by default, _especially if used as a texture_, but as long as the docs are crystal clear I'm not sure it's a big deal either way.

**Comment by ivanpopelyshev** [comment]
@photonstorm Huzzah! At last.

**Comment by GoodBoyDigital** [comment]
@glebmachine  - looks like there is something wrong with this PR. Fancy posting this again to dev? cheers!

**Pull request closed without merging**